### PR TITLE
[1043] change course incremental api to use sub-second changed_since

### PR DESCRIPTION
**Pull request by misaka**
This brings it inline with the implementation of the provider incremental api
endpoint. Changes include:

- remove from_course_id param from next link header and Course scope
- format changed_since param in next link header to include micro-seconds

### Guidance to review

We want to add some more thorough tests that exercise larger page sizes and counts, it seems the existing tests didn't catch certain failures, but the implementation seems solid and can be reviewed independently of those tests.

We plan to make another PR to use `changed_at` instead of `updated_at`, which is required to record updates done to course enrichments.

### CODE REVIEW

File: spec/controllers/api/v1/courses_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      let(:changed_since) { 10.minutes.ago.utc }
+
+      before do
+        controller.index
+      end
+
+      describe 'returned courses in JSON' do
+        subject { response.body }
+
+        # TODO: This isn't very clear, it should be "should_not have_courses"
```
Comment (benilovj) [inline spec/controllers/api/v1/courses_controller_spec.rb]: ?? is this an issue with the matcher?

File: spec/models/course_spec.rb [hunk 2]
```diff
@@ -11,8 +11,8 @@
 #  qualification           :integer          not null
 #  start_date              :datetime
 #  study_mode              :text
-#  accrediting_provider_id :integer
-#  provider_id             :integer          default(0), not null
+#  accrediting_course_id :integer
+#  course_id             :integer          default(0), not null
```
Comment (benilovj) [inline spec/models/course_spec.rb]: what are these changes related to?

File: spec/controllers/api/v1/courses_controller_spec.rb (hunk 1 above)
Comment (misaka) [reply spec/controllers/api/v1/courses_controller_spec.rb]: I think there should be some way to get the matcher to behave better in this instance, just haven't looked yet.

File: spec/models/course_spec.rb (hunk 2 above)
Comment (misaka) [reply spec/models/course_spec.rb]: I think this was brought in by a search/replace. Good spot, will regenerate the annotations.

File: app/models/course.rb [hunk 3]
```diff
@@ -53,10 +53,12 @@ class Course < ApplicationRecord
...
 
-  scope :changed_since, ->(datetime, from_course_id = 0) do
-    if datetime.present?
-      where("course.updated_at >= ? AND course.id > ?", datetime, from_course_id).order(:updated_at, :id)
-    end
+  scope :changed_since, ->(timestamp) do
+    if timestamp.present?
+      where("course.updated_at > ?", timestamp)
+    else
+      where("updated_at is not null")
```
Comment (benilovj) [inline app/models/course.rb]: would `where.not(updated_at: nil)` be more idiomatic?

Comment (timabell) [reply app/models/course.rb]: done, tests still pass, cheers

**Comment by timabell** [comment]
> I think the API docs still has a reference to `from_course_id`

done @benilovj

File: spec/controllers/api/v1/courses_controller_spec.rb (hunk 1 above)
Comment (timabell) [reply spec/controllers/api/v1/courses_controller_spec.rb]: suggested improvement: https://github.com/DFE-Digital/manage-courses-backend/pull/154/commits/bbbf446a8bd99c36e462044442061a49d3c11918
thoughts @misaka @benilovj  ?
